### PR TITLE
fix: fetch payment terms template from sales order when creating sales invoice from delivery note

### PR DESCRIPTION
**Pull request by dj12djdjs**
### Expected Result
1. Create Sales Order
2. Modify Payment Terms Template
3. Create Delivery Note
4. Create Sales Invoice
5. Expected Sales Invoice Payment Terms Template to Match Sales Order Payment Terms Template

### Actual Result
1. Create Sales Order
2. Modify Payment Terms Template
3. Create Delivery Note
4. Create Sales Invoice
5. Sales Invoice Payment Terms Template is set to Customer `payment_terms` NOT the Payment Terms Template set in Sales Order

### CODE REVIEW

File: erpnext/stock/doctype/delivery_note/delivery_note.py [hunk 1]
```diff
@@ -657,6 +661,24 @@ def make_sales_invoice(source_name, target_doc=None):
 	invoiced_qty_map = get_invoiced_qty_map(source_name)
 
 	def set_missing_values(source, target):
+		try:
+			parent_payment_term_templates = frappe.get_all(
+				"Sales Order",
+				{
+					"name": ["IN", [d.against_sales_order for d in doc.items]],
```
Comment (dj12djdjs) [inline erpnext/stock/doctype/delivery_note/delivery_note.py:668]: I've been trying to test use cases where d.against_sales_order is NULL in all rows. Expecting pymysql to throw an exception. But seems this is being handled somewhere in frappe maybe?

**Comment by deepeshgarg007** [comment]
keep-alive

**Comment by dj12djdjs** [comment]
@rohitwaghchaure @s-aga-r

**Comment by s-aga-r** [comment]
@deepeshgarg007

File: erpnext/stock/doctype/delivery_note/delivery_note.py (hunk 1 above)
Comment (s-aga-r) [reply erpnext/stock/doctype/delivery_note/delivery_note.py:668]: The below query will be generated if there is no `against_sales_order`.

```
select
    `payment_terms_template`
from
    `tabSales Order`
where
    coalesce(`tabSales Order`.`name`, '') IN ('')
order by
    `tabSales Order`.docstatus asc,
    `tabSales Order`.`modified` DESC
```

A check before would be good:

```
against_sales_orders = [d.against_sales_order for d in doc.items if d.against_sales_order]

if against_sales_orders:
    parent_payment_term_templates = frappe.get_all(...
```

**Comment by deepeshgarg007** [comment]
Screenshot 2023-08-25 at 8 49 43 PM

@dj12djdjs There is settings for this already in Accounts Settings, please check this once and reopen the PR if this doesn't work for you

**Pull request closed without merging**